mikupad.html

```diff
@@ -1277,7 +1277,15 @@
 	const memoryArea = useRef();
 
 	const promptText = useMemo(() => joinPrompt(promptChunks), [promptChunks]);
-	const modifiedPrompt = useMemo(() => memoryTokens.concat(promptText.substring(Math.round(promptText.length-(contextLength*3.3) + (memoryTokens.length/3.3))+1)), [contextLength, promptText, memoryTokens]);
+	const modifiedPrompt = useMemo(() => 
+		memoryTokens.concat(
+			promptText.substring(
+				Math.round(
+					promptText.length-(contextLength*3.3) + (memoryTokens.length/3.3)
+				)+1
+			)
+		), 

```

Comment:
I didn't quite mean that with my review, to be honest. What I had in mind was more like this:
```js
() => {
	const tokenRatio = 3.3;
	const estimatedPromptLength = Math.round(promptText.length - contextLength * tokenRatio + memoryTokens.length / tokenRatio) + 1;
	return memoryTokens.concat(promptText.substring(estimatedPromptLength));
}
```
(If you don't have a better idea, feel free to use the code above)